File: CookieClickerGame/Score.cs
using Akka.Actor;
using Akka.Util;
using System;
using System.Collections.Generic;
using System.Text;

namespace CookieClickerGame
{
    public class Score : ReceiveActor
    {
        #region
        public class Increase { }
        public class Decrease
        {
            public int Amount { get; set; }

            public Decrease(int amount)
            {
                Amount = amount;
            }
        }
        public class GetScore
        {
        }
        #endregion

        private int score;

        public Score()
        {
            Receive<Increase>(HandleIncrease);
            Receive<Decrease>(HandleDecrease);
            Receive<GetScore>(HandleGetScore);
        }

        private void HandleGetScore(GetScore obj)
        {
            Sender.Tell(new Result<int>(score));
        }

        private void HandleDecrease(Decrease obj)
        {
            if(obj.Amount <= score)
            {
                score -= obj.Amount;
                Sender.Tell(new Result<bool>(true));
            }

            Sender.Tell(new Result<bool>(false));
        }

        private void HandleIncrease(Increase increase)
        {
            score++;
        }
    }


}

File: scala/cookieclicker/src/main/scala/com/example/Score.scala
package cookieclicker

import akka.actor.{Actor}

class Score extends Actor {
    import Score._
  override def receive: Receive = behaviors(0)

  def behaviors(score: Int): Receive = {
    case Increase => context.become(behaviors(score + 1))
    case Decrease(amount) => 
        if (amount <= score) {
            sender() ! true
            context.become(behaviors(score - amount))
        } else {
            sender() ! false
        }
    case GetScore => sender() ! score
  }
}

object Score {
    case class Increase()
    case class Decrease(amount: Int)
    case class GetScore()
}
